src/common.cc

```diff
@@ -324,4 +324,18 @@ namespace sharp {
     }
   }
 
+  /*
+    Get VIPS Boolean operatoin type from string
+  */
+  VipsOperationBoolean GetBooleanOperation(std::string opStr) {
+    if(opStr == "and" ) {
```

Comment:
The rather handy `vips_enum_from_nick` function should allow this to be simplified:
```cpp
return static_cast<VipsOperationBoolean>(
  vips_enum_from_nick(nullptr, VIPS_TYPE_OPERATION_BOOLEAN, opStr.data())
);
```